### PR TITLE
update — Devito 4.2.3

### PR DESCRIPTION
**Pull request by alisiahkoohi**
This pull request updates the source code and test to make them compatible with the latest changes in Devito

### CODE REVIEW

File: devito4pytorch/devito_wrapper.py [hunk 1]
```diff
@@ -29,8 +33,8 @@ def backward(ctx, grad_output):
         rec.data[:] = grad_output[:]
 
         # Adjoint linearized modeling
-        u0 = ctx.solver.forward(save=True)[1]
-        g = ctx.solver.gradient(rec, u0)[0].data
+        u0 = ctx.solver.forward(save=True, vp=ctx.model.vp)[1]
```
Comment (mloubout) [inline devito4pytorch/devito_wrapper.py]: why does this one need vp?

File: devito4pytorch/devito_wrapper.py [hunk 2]
```diff
@@ -54,8 +60,8 @@ def forward(ctx, input, model, geometry, solver, device):
         rec = ctx.geometry.rec
         rec.data[:] = input[:]
 
-        u0 = ctx.solver.forward(save=True)[1]
-        g = ctx.solver.gradient(rec, u=u0)[0].data
+        u0 = ctx.solver.forward(save=True, vp=ctx.model.vp)[1]
```
Comment (mloubout) [inline devito4pytorch/devito_wrapper.py]: same

File: tests/adjoint_born_test.py [hunk 3]
```diff
@@ -14,13 +14,17 @@
...
+    """
     def __init__(self, model, geometry, device):
         super(AdjointBornLayer, self).__init__()
         self.forward_born = devito_wrapper.AdjointBorn()
         self.model = model
         self.geometry = geometry
         self.device = device
-        self.solver = AcousticWaveSolver(self.model, self.geometry, space_order=8)
+        self.solver = AcousticWaveSolver(self.model, self.geometry,
+                                         space_order=4)
```
Comment (mloubout) [inline tests/adjoint_born_test.py]: bit low, would keep 8 to keep dispersion fairly low

File: tests/adjoint_born_test.py [hunk 4]
```diff
@@ -31,29 +35,34 @@ def test_adjoint_born():
...
     geometry0 = setup_geometry(model0, tn)
 
     # Pure Devito
-    solver0 = AcousticWaveSolver(model0, geometry0, space_order=8)
-    dm = (model.vp.data**(-2) - model0.vp.data**(-2))
+    solver0 = AcousticWaveSolver(model0, geometry0, space_order=4)
+    dm = model.vp.data**(-2) - model0.vp.data**(-2)
 
-    grad_devito = np.array(solver0.born(-dm)[0].data)
+    grad_devito = np.array(solver0.jacobian(-dm)[0].data)
```
Comment (mloubout) [inline tests/adjoint_born_test.py]: would put `dm = model0.vp.data**(-2) - model.vp.data**(-2)` and `jacobian(dm)` toavoid that minus sign, may make people think that gradient sign is wrong.

**Review by mloubout — COMMENTED** [review]
Few minor comments but looks good to me

File: devito4pytorch/devito_wrapper.py (hunk 1 above)
Comment (alisiahkoohi) [reply devito4pytorch/devito_wrapper.py]: I was unsure about this. Does not really. Will fix other instances of it as well.

File: devito4pytorch/devito_wrapper.py (hunk 2 above)
Comment (alisiahkoohi) [reply devito4pytorch/devito_wrapper.py]: Thanks.

File: tests/adjoint_born_test.py (hunk 3 above)
Comment (alisiahkoohi) [reply tests/adjoint_born_test.py]: Okay.

File: tests/adjoint_born_test.py (hunk 4 above)
Comment (alisiahkoohi) [reply tests/adjoint_born_test.py]: Alright, that is more reasonable.

Comment (alisiahkoohi) [reply tests/adjoint_born_test.py]: Regardless of this, we want to compare linearized data w/ the following gradient (computed via pytorch) at `d=0`:



We need a minus sign somewhere.

Comment (mloubout) [reply tests/adjoint_born_test.py]: That's an original gradient :D  but yes makes sense then. 
Maybe add a comment or docstring for it? Feels weird to see that minus sign without it? but gtg

Comment (alisiahkoohi) [reply tests/adjoint_born_test.py]: Yeah it is a gradient just for the sake of testing. I moved that minus sign to few lines later with a comment stating that we need the negative gradient of the loss function to get linearized data. 

Thanks will merge.
